opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -93,26 +103,47 @@
 
         /**
          * A shorthand for addReference(References.CHILD_OF, parent).
+         *
+         * If parent==null, this is a noop.
          */
         SpanBuilder asChildOf(SpanContext parent);
 
         /**
          * A shorthand for addReference(References.CHILD_OF, parent.context()).
+         *
+         * If parent==null, this is a noop.
          */
-        SpanBuilder asChildOf(Span parent);
+        SpanBuilder asChildOf(BaseSpan parent);
```

Comment:
any particular reason for using a raw type here instead of something like `<T extends BaseSpan<?>> SpanBuilder asChildOf(BaseSpan<T> parent)`?